lib/game.rb

```diff
@@ -0,0 +1,81 @@
+require 'pry'
+class Game
+
+  attr_accessor :board, :player_1, :player_2, :winner
+
+  WIN_COMBINATIONS = [[0,1,2],
+                     [3,4,5],
+                     [6,7,8],
+                     [0,3,6],
+                     [1,4,7],
+                     [2,5,8],
+                     [0,4,8],
+                     [6,4,2]]
+
+  def initialize (player_1=Players::Human.new("X"), player_2=Players::Human.new("O"), board=Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+    @board.display
+    @winner = winner
+  end
+
+  def start
+    puts "Welcome to TTT with AI!"
+    play
+    @board.display
+  end
+
+  def current_player
+    @board.turn_count%2 == 0 ? @player_1 : @player_2
+  end
+
+  def over?
+    draw? || won?
+  end
+
+  def won?
```

Comment:
Here you can use [`Enumerable#detect`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-detect) to return the winning combination or `nil` if none is found.

```ruby
  def won?
    WIN_COMBINATIONS.detect do |combo|
      @board.cells[combo[0]] == @board.cells[combo[1]] &&
      @board.cells[combo[0]] == @board.cells[combo[2]] &&
      @board.taken?(combo[0] + 1)
    end
  end
```